### PR TITLE
Feature/vimeo omniauth

### LINKED ISSUE #1782: Add support for Vimeo publishers
Create channel for Vimeo publishers.
At launch, the browser will support tipping and auto-contributions in terms of payout.

This is a P2; as in if there are no fires, this needs to get prioritized.

cc @yachtcaptain23

### PR DESCRIPTION
**Pull request by yachtcaptain23**
## Pull Request Name
Vimeo Feature

#### Features
Closes #1782 

#### Pull Request Checklist

- [ ] Adequate test coverage exists to prevent regressions -- [Guide to Testing](https://guides.rubyonrails.org/testing.html)
- [ ] XSS is mitigated -- [Guide to XSS Prevention](https://guides.rubyonrails.org/security.html#cross-site-scripting-xss)
- [ ] No raw SQL -- [Guide to SQL Injection Prevention](https://guides.rubyonrails.org/security.html#sql-injection)
- [ ] UI/UX is responsive -- [Guide to Responsive UI/UX](https://developers.google.com/web/fundamentals/design-and-ux/responsive/)
- [ ] Integrated Matomo for new UI elements -- [Guide to Matomo](https://developer.matomo.org/guides/integrate-introduction)
- [ ] Passes checklist for Progressive Web App -- [Guide to PWAs](https://developers.google.com/web/progressive-web-apps/checklist)

### CODE REVIEW

File: app/controllers/publishers_controller.rb [hunk 1]
```diff
@@ -162,10 +162,6 @@ def confirm_two_factor_authentication_removal
     end
   end
 
-  def javascript_detected
```
Comment (corymcdonald) [inline app/controllers/publishers_controller.rb]: 🎉 we're finally removing this ❤️

File: app/controllers/publishers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -157,6 +157,31 @@ def register_twitter_channel
...
+      @channel.details = VimeoChannelDetails.new(
+        name: vimeo_auth_hash.info.name,
+        vimeo_channel_id: vimeo_auth_hash.info.id,
+        auth_provider: vimeo_auth_hash.info.auth_provider,
+        thumbnail_url: vimeo_auth_hash.info.pictures.last.link,
+        channel_url: vimeo_auth_hash.info.link,
+        nickname: vimeo_auth_hash.info.nickname
+      )
+
+      existing_channel = Channel.joins(:vimeo_channel_details).where("vimeo_channel_details.vimeo_channel_id": vimeo_auth_hash.info.id).first
```
Comment (corymcdonald) [inline app/controllers/publishers/omniauth_callbacks_controller.rb]: Let's make sure the existing vimeo channel that we're trying to find is verified

File: app/models/vimeo_channel_details.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  validates :thumbnail_url, presence: true
+  validates :name, presence: true
+  validates :channel_url, presence: true
+
+  def channel_identifier
+    "#{VIMEO_PREFIX}#{vimeo_channel_id}"
+  end
+
+  def omniauth_url
+    "https://api.vimeo.com/oauth/authorize?response_type=code&client_id=7ca223a095b8d2b2aecb9d1317b5af7b5c4c70f3&redirect_uri=https://localhost:3000/publishers/home&state=1"
```
Comment (corymcdonald) [inline app/models/vimeo_channel_details.rb]: We'll want to move this to a ENV variable, but that can be done later

Comment (yachtcaptain23) [reply app/models/vimeo_channel_details.rb]: I'll be following up with devops to get this account setup

File: app/views/publishers/_choose_channel_type.html.slim [hunk 4]
```diff
@@ -38,3 +38,12 @@
         .icon-and-text--text
           h6= t "publishers.choose_new_channel_type.twitter.heading"
           = t "publishers.choose_new_channel_type.twitter.description"
+
+= link_to(publisher_register_vimeo_channel_omniauth_authorize_path, class: 'card block-link--card', :"data-piwik-action" => "AddVimeoChannelClicked", :"data-piwik-name" => "Clicked", :"data-piwik-value" => "AddChannelFlow") do
+  .card-body
+    .icon-and-text--wrapper
+      .icon-and-text--icon
+        = image_tag "choose-channel/choose-channel--vimeo.jpg", height: 72, width: 88, class: "choose-channel--icon"
```
Comment (corymcdonald) [inline app/views/publishers/_choose_channel_type.html.slim]: I don't think we should specify a width because right now it's scaled incorrectly

**Comment by corymcdonald** [comment]
Need to add the vimeo channel details in the below files as well
`/app/services/promo/publisher_channels_registrar.rb`
`/app/controllers/api/v1/stats/channels_controller.rb`

**Comment by corymcdonald** [comment]
Closing in favor of #1881

**Pull request closed without merging**